lib/src/code/code.dart

```diff
@@ -92,15 +95,32 @@ class Code {
         const [];
     final sectionsMap = {for (final s in sections) s.name: s};
 
-    _applyNamedSectionsToLines(
-      lines: lines.lines,
-      sections: sectionsMap,
-      readOnlySectionNames: readOnlySectionNames,
+    final isCodeReadonly = visibleSectionsNames.isNotEmpty;
+    if (isCodeReadonly) {
+      _makeCodeReadonly(lines: lines.lines);
+    } else {
+      _applyReadonlySectionsToLines(
+        lines: lines.lines,
+        sections: sectionsMap,
+        readOnlySectionNames: readOnlySectionNames,
+      );
+    }
+
+    final visibleSectionsHiddenRanges = _visibleSectionsToHiddenRanges(
+      visibleSectionsNames,
+      sectionsMap,
+      lines,
+    );
+
+    final commentsHiddenRanges = _commentsToHiddenRanges(
+      serviceComments,
+      visibleSectionsHiddenRanges,
```

Comment:
Revert, the builder can merge overlapping ranges.

Reply:
+